docs(developer): add developer docs and refactor README

chalin opened the pull request:
A first edition of `DEVELOPER.md` inspired from the angular.dart version, the angular.js documentation, and expanded to include instructions given in this repos’ `README.md` (which has been cleaned up and padded up).

vicb commented on DEVELOPER.md:
```diff
@@ -0,0 +1,233 @@
+  Dartium (a version of [Chromium](http://www.chromium.org) with native support for Dart through
+  the Dart VM). One of the **simplest** ways to get both is to install the **Dart Editor bundle**,
+  which includes the editor, SDK and Dartium. See the [Dart tools download page for
+  instructions](https://www.dartlang.org/tools/download.html).
```
may be link http://gsdview.appspot.com/dart-archive/channels/dev/release/latest/ to download dev releases

mhevery replied:
Also make it clear that it is optional

chalin replied:
Actually, the download page that is being linked on line 27 is inclusive. I.e., it also provides a link to dev channel versions (via the archive page):
> Want another Dart version? Use the [download archive](https://www.dartlang.org/tools/download-archive). It lets you choose specific versions of the Dart downloads. It also has the latest **dev channel** builds, which let you try new features a few weeks earlier than the stable channel.

And the interface provided by the archive page is much nicer than gsdview.appspot -- see below. **I would avoid adding the extra link here. Agreed?**

-----

![image](https://cloud.githubusercontent.com/assets/4140793/6644935/4baa8c76-c975-11e4-8cdf-0faea7dce000.png)

vicb replied:
Cool! I was not aware about this. I always have troubles finding the dev versions which is why I have a bookmark on gsdview but your link is much better!

mhevery commented:
Love it! Thanks for helping others to get up to speed!

chalin commented:
Ok then, anything else to be done or can this be merged? (Build is failing but I can't imagine it is due to my commit.)

chalin committed:
docs(developer): add developer docs and refactor README

A first edition of `DEVELOPER.md` inspired from the angular.dart version
and expanded to include instructions given in this repos’ `README.md`
(which has been cleaned up and padded up).

chalin commented:
Since the question came up again (in #982), I added an explicit mention of the stable and dev streams being available from the download archive.
